messages.rb

```diff
@@ -505,6 +506,84 @@ def send_diff(event)
   event << "Score changes on #{current.format_name} since #{since}:\n```#{diff}```"
 end
 
+def do_analysis(scores, rank)
+  run = scores.download_scores(rank)
+  if !run.nil?
+    player = run['user_name']
+    replay_id = run['replay_id'].to_s
+    score = "%.3f" % [run['score'].to_f / 1000]
+    analysis = scores.analyze_replay(replay_id)
+    gold = "%.0f" % [((run['score'].to_f / 1000) + (analysis.size.to_f / 60) - 90) / 2]
+    result = {'player' => player, 'scores' => scores.format_name, 'rank' => rank, 'score' => score, 'analysis' => analysis, 'gold' => gold}
+    return result
+  else
+    return nil
```

Comment:
Generally the style in here has been to early-out (and I prefer this as a general rule) - so instead of indenting your whole method inside this `if` block, and forcing the 'early-return' logic to the bottom which is logically a bit backwards, make the second line `return nil if run.nil?`, and then the rest of the logic is the 'main chunk' of the method.

Reply:
Wow you're right, can't believe I had such unclean implementation.